src/NuGet.Core/NuGet.ProjectModel/PackageSpecUtility.cs

```diff
@@ -101,9 +114,23 @@ public static bool IsSnapshotVersion(string version)
                 throw new ArgumentNullException(nameof(projectFramework));
             }
 
+            var hasImports = imports?.Any() == true;
```

Comment:
Same as other place, get a list out of Imports and use it through out instead of generating it multiple times.